### PR TITLE
[prometheus] Improve handling of extra args

### PR DESCRIPTION
**Pull request by h0tbird**
#### Which issue this PR fixes

This PR enables the inclusion of additional arguments like `--web.enable-remote-write-receiver`, which do not support specifying a value.

**Before**
```console
$ helm template charts/prometheus --set 'server.extraArgs.web\.enable-remote-write-receiver=' | grep remote
            - --web.enable-remote-write-receiver=
```

Notice the trailing `=` with no value, this 👆🏼 breaks prometheus.

**After**
```console
$ helm template charts/prometheus --set 'server.extraArgs.web\.enable-remote-write-receiver=' | grep remote
            - --web.enable-remote-write-receiver
```

Without the trailing `=`, prometheus is happy.

#### Checklist

- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

**Review by zeritti — CHANGES_REQUESTED** [review]
Thank you, @h0tbird, for your PR. Please, see my comments below.

File: charts/prometheus/templates/deploy.yaml [hunk 1]
```diff
@@ -75,7 +75,11 @@ spec:
           {{- end }}
             - --reload-url={{ default $default_url .Values.configmapReload.reloadUrl }}
           {{- range $key, $value := .Values.configmapReload.prometheus.extraArgs }}
+            {{- if eq (toString $value) "" }}
+            - --{{ $key }}
+            {{- else }}
             - --{{ $key }}={{ $value }}
+            {{- end }}
```
[the remark this answers is not in the document]
Comment (h0tbird) [reply charts/prometheus/templates/deploy.yaml]: Thank you! This approach is more elegant 👍🏼
With `helm template --set`, I still have to run it with a trailing `=`:
```console
$ helm template charts/prometheus --set 'server.extraArgs.web\.enable-remote-write-receiver=' | grep remote
            - --web.enable-remote-write-receiver
```
It is fine, not sure if there is a better way 🤔

File: charts/prometheus/templates/deploy.yaml [hunk 2]
```diff
@@ -144,7 +148,11 @@ spec:
             - --{{ . }}
           {{- end }}
           {{- range $key, $value := .Values.server.extraArgs }}
+            {{- if eq (toString $value) "" }}
+            - --{{ $key }}
+            {{- else }}
             - --{{ $key }}={{ $value }}
+            {{- end }}
```
[the remark this answers is not in the document]
Comment (h0tbird) [reply charts/prometheus/templates/deploy.yaml]: Ouch! I overlook `server.extraFlags`, this works for me too:
```console
$ helm template charts/prometheus --set 'server.extraFlags={web.enable-remote-write-receiver}' | grep remote
            - --web.enable-remote-write-receiver
```

**Review by zeritti — APPROVED** [review]
Thank you, @h0tbird, LGTM.

**Pull request merged**